models/meal-plan.model.js

```diff
@@ -24,6 +24,16 @@ const mealPlanSchema = Schema(
       data: Buffer,
       contentType: String,
     },
+    isFavourite: {
+      type: Boolean,
+      default: false,
+    },
+    favouriteCount: {
+      type: Number,
+      default: 0,
+      min: 0,
+    },
+    likes: [{ type: Schema.Types.ObjectId, ref: "User" }],
```

Comment:
do we need these two? 
I think we just need likes array and then we will be able to calculate favouritesCount. (remove favouriteCount)